clouddriver-ecs/src/test/groovy/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProviderSpec.groovy

```diff
@@ -110,9 +114,127 @@ class EcsServerClusterProviderSpec extends Specification {
     def ec2Instance = new Instance(
       placement: new Placement(
         availabilityZone: availabilityZone
+      ),
+      vpcId: 'vpc-1234',
+      securityGroups: [new GroupIdentifier (
+        groupId: 'sg-1234'
+      )]
+    )
+
+    def taskDefinition = new TaskDefinition(
+      containerDefinitions: [
+        new ContainerDefinition(
+          image: 'my-image',
+          memoryReservation: 256,
+          cpu: 123,
+          environment: [],
+          portMappings: [new PortMapping(containerPort: 1337)]
+        )
+      ]
+    )
+
+    def scalableTarget = new ScalableTarget(
+      minCapacity: 1,
+      maxCapacity: 2,
+      resourceId: "service:/mycluster/${serviceName}"
+    )
+
+    def ecsServerGroupEast = makeEcsServerGroup(serviceName, 'us-east-1', startedAt.getTime(), taskId, healthStatus, ip)
+    def ecsServerGroupWest = makeEcsServerGroup(serviceName, 'us-west-1', startedAt.getTime(), taskId, healthStatus, ip)
+
+    def expectedCluster = new EcsServerCluster()
+    expectedCluster.setAccountName(creds.getName())
+    expectedCluster.setName(familyName)
+    expectedCluster.setServerGroups(new HashSet([ecsServerGroupEast, ecsServerGroupWest]))
+    expectedCluster.setLoadBalancers(Collections.singleton(loadbalancer))
+
+
+    def serviceAttributes = ServiceCachingAgent.convertServiceToAttributes(creds.getName(), creds.getRegions()[0].getName(), cachedService)
+    def taskAttributes = TaskCachingAgent.convertTaskToAttributes(task)
+
+    def serviceCacheData = new DefaultCacheData('', serviceAttributes, [:])
+    def taskCacheData = new DefaultCacheData('', taskAttributes, [:])
+
+    accountCredentialsProvider.getAll() >> [creds]
+    ecsLoadbalancerCacheClient.find(_, _) >> [loadbalancer]
+    containerInformationService.getTaskPrivateAddress(_, _, _) >> "${ip}:1337"
+    containerInformationService.getHealthStatus(_, _, _, _) >> [healthStatus]
+    containerInformationService.getEc2Instance(_, _, _) >> ec2Instance
+    taskDefinitionCacheClient.get(_) >> taskDefinition
+    scalableTargetCacheClient.get(_) >> scalableTarget
+    ecsCloudWatchAlarmCacheClient.getMetricAlarms(_, _, _) >> []
+
+    cacheView.filterIdentifiers(_, _) >> ['key']
+    cacheView.getAll(Keys.Namespace.SERVICES.ns, _) >> [serviceCacheData]
+    cacheView.getAll(Keys.Namespace.TASKS.ns, _) >> [taskCacheData]
+
+    when:
+    def retrievedCluster = provider.getCluster("myapp", creds.getName(), familyName)
+
+    then:
+    retrievedCluster == expectedCluster
+  }
+
+  def 'should produce an ecs cluster with VPC network configuration'() {
+    given:
+    def applicationName = 'myapp'
```

Comment:
consider moving some of these duplicate fields outside the spec, using a `setupSpec()` block or combining the two tests with `where`. It tends to improve the readability of the test when they are more concise.